SeaBattle/BaseShip.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Reflection;
+using System.Xml.Linq;
+using System.Data.SqlTypes;
+using System.Runtime.CompilerServices;
+
+namespace SeaBattle
+{
+    public class BaseShip
+    { 
+        public List<Deck> Decks { get; set; }
+        private int Durability { get; set; }
+        private int MoveSpeed { get; set; }
+        private int MoveDirectionX { get; set; }
+        private int MoveDirectionY { get; set; }
+        private string Name { get; set; }
+        private int Speed { get; set; }
+        
+        public BaseShip(string Name, List<Point> points) {
+            var Decks = new List<Deck>();
+
+            for (int i = 0; i < points.Count - 1; i++)
+            {
+                Deck a = new Deck();
+                a.Location = points[i];
+                Decks.Add(a);
+            }
+        }
+        public override string ToString()
+        {
+            StringBuilder output = new StringBuilder ("Name: ", 25);
+            output.Append (Name + ", ");
+            output.Append("Type: " + this.GetType() + ", ");
+            output.Append("Durability: " + Durability + ", ");
+            output.Append("Speed: " + Speed + ", ");
+            output.Append("Decks:" + Decks.Count + ", ");
+            output.Append("Coordinates: ");
+            foreach (var deck in Decks)
+            {
+                output.Append(deck.Location.X + ":" + deck.Location.Y + ", ");
+            }
+            return output.ToString();
+        }
+        public static bool operator== (BaseShip s1, BaseShip s2)
+        {
+            return ((s1.Decks.Count.Equals(s2.Decks.Count)) && (s1.GetType().Equals(s2.GetType())));
+        }
+        public static bool operator!= (BaseShip s1, BaseShip s2)
+        {  
+            return !((s1.Decks.Count.Equals(s2.Decks.Count)) && (s1.GetType().Equals(s2.GetType())));
+
+        }
+        public bool Equals (BaseShip s1)
+        {
+            return ((this.Decks.Count == s1.Decks.Count) && (this.GetType() == s1.GetType()));
+
+        }
+
+        // This function adds s1(BaseShip) to the Sea
+        // 1. + Checking if Point within Sea borders
+        // 2. + Checking if Point is available 
+        public void AddShip(BaseShip s1, Sea PlaceSea)

```

Comment:
Yes as I said, I think Sea is passive and ships doing all actions.

Reply:
I don't think so, you can split responsibility and put the logic of adding the ship to the Sea, it will be much more clear and native in my opinion than just putting everything in the Ship class. It's not about the passiveness of the entity during the game or smth, it's about logic, readability, and convenience.

Reply:
It's native when the class where you put stuff is responsible for the logic, and how to do it rather than class put stuff inside other class and handles how to do it

Reply:
Moved Addship method to the Sea.